examples/llava/minicpmv_wrapper.cpp

```diff
@@ -0,0 +1,145 @@
+#include "ggml.h"
+#include "common.h"
+#include "clip.h"
+#include "llava.h"
+#include "minicpmv_wrapper.h"
+#include "llama.h"
+#include <cstdio>
+#include <cstdlib>
+#include <vector>
+
+struct llama_model * llava_init(gpt_params * params) {
+    llama_backend_init();
+    llama_numa_init(params->numa);
+
+    llama_model_params model_params = llama_model_params_from_gpt_params(*params);
+
+    llama_model * model = llama_load_model_from_file(params->model.c_str(), model_params);
+    if (model == NULL) {
+        LOG_TEE("%s: error: unable to load model\n" , __func__);
+        return NULL;
+    }
+    return model;
+}
+
+struct minicpmv_context * llava_init_context(gpt_params * params, llama_model * model) {
+    auto prompt = params->prompt;
+    if (prompt.empty()) {
+        prompt = "describe the image in detail.";
+    }
+
+    llama_context_params ctx_params = llama_context_params_from_gpt_params(*params);
+    ctx_params.n_ctx           = params->n_ctx < 2048 ? 2048 : params->n_ctx; // we need a longer context size to process image embeddings
```

Comment:
(nitpick) We should warn the user that we expand the context.

Reply:
I think this is the simplest way

```cpp
ctx_params.nctx = params->n_ctx;
if (params->n_ctx < 2048) {
    // warn user
    ctx_params.n_ctx = 2048;
}
```

alternatively
```cpp
if (params->n_ctx < 2048) {
    // warn user here, "Image processing requires at least 2048 context, setting context to 2048"
    ctx_params.n_ctx = 2048;
} else {
    ctx_params.n_ctx = params->n_ctx;
}
```

Reply:
[dad4abe](https://github.com/ggerganov/llama.cpp/pull/7599/commits/dad4abe1bccb31065b6f6d4736181ff423a34d5c)
Do you think it's okay to write like this?